let DEBUG_MODE,ValidationError,CustorErrorHandler;_589‍.x([["default",()=>_589‍.o]]);_589‍.w("../config",[["DEBUG_MODE",["DEBUG_MODE"],function(v){DEBUG_MODE=v}]]);_589‍.w("joi",[["ValidationError",["ValidationError"],function(v){ValidationError=v}]]);_589‍.w("../services/customErrorHandler",[["default",["CustorErrorHandler"],function(v){CustorErrorHandler=v}]]);



const errorHandler = (err, req, res, next) => {
    let statusCode = 500;

    let data = {
        message: "Internal Server Error",
        ...(DEBUG_MODE === 'true' && { originalError: err.message })
    }

    if (err instanceof ValidationError) {
        statusCode = 422;
        data = {
            message: err.message
        }
    }

    if (err instanceof CustorErrorHandler) {
        statusCode = err.status;
        data = { message: err.message }
    }

    return res.status(statusCode).json(data);
}

_589‍.d(errorHandler);
